### PR TITLE
Configure packaging to include multi-release classes

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by dwijnand** [comment]
test failed, but then so is upstream #23651.

**Comment by ktoso** [comment]
Yeah validation will be failing here, ignore it :)

**Comment by ktoso** [comment]
(this branch is currently "weird")

File: project/OSGi.scala [hunk 1]
```diff
@@ -23,7 +23,8 @@ object OSGi {
     OsgiKeys.privatePackage := Seq())
 
   val actor = osgiSettings ++ Seq(
-    OsgiKeys.exportPackage := Seq("akka*"),
+    OsgiKeys.bundle := OsgiKeys.bundle.dependsOn(compile in MultiReleaseJar).value,
+    OsgiKeys.exportPackage := Seq("akka*", "META-INF.versions.9.akka*"), // TODO: Refine this as it errors
```
Comment (ktoso) [inline project/OSGi.scala]: whoa that's weird; if it makes things work then cool I guess

File: project/plugins.sbt [hunk 2]
```diff
@@ -23,10 +23,9 @@ addSbtPlugin("com.lightbend.akka" % "sbt-paradox-akka" % "0.4")
 addSbtPlugin("com.lightbend" % "sbt-whitesource" % "0.1.6")
 addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.9.3")
 
-// // JDK9 features support via Multi-Release JARs
-// addSbtPlugin("com.lightbend.sbt" % "sbt-multi-release-jar" % "0.1.0") // we don't actually use it so far
+// JDK9 features support via Multi-Release JARs
+addSbtPlugin("com.lightbend.sbt" % "sbt-multi-release-jar" % "0.1.0")
```
Comment (ktoso) [inline project/plugins.sbt]: could be 1.0.2

File: project/OSGi.scala (hunk 1 above)
Comment (dwijnand) [reply project/OSGi.scala]: This is a bit crude. It puts the classfile in the right place in the jar. But from an OSGi bundle standpoint it's probably not legal - thus the non-fatal warning it spits out. Up to you if this is a blocker.

**Comment by dwijnand** [comment]
review and/or merge?

**Comment by ktoso** [comment]
I checked this and it seems to work indeed, thanks whole a lot Dale!

For reference, it does trigger these scary messages:

```

[info] Validating all packages are set private or exported for OSGi explicitly...
[warn] bnd: Unused Private-Package instructions, no such package(s) on the class path: [akka.osgi.impl]
[warn] bnd: Unused Export-Package instructions: [META-INF.versions.9.akka*]
[error] bnd: Classes found in the wrong directory: {META-INF/versions/9/akka/util/Unsafe.class=akka.util.Unsafe}
```

But the produced manifest (the osgi stuff) does look fine anyway.

It seems it LOST the `Multi-Release: true` however, so it won't still work... It definitely is a good step though in the right direction so LGTM, though we have to look into it still.

**Comment by ktoso** [comment]
The osgi thing seems to break the manifest info about `Multi-Release` it seems...
I even tried doing `set  packageOptions in ThisBuild +=  Package.ManifestAttributes( "Multi-Release" -> "true" )` but it would not include that value in the resulting manifest hm...

**Comment by dwijnand** [comment]
Hmm, I'll have a look tomorrow morning (UTC+1), see if I can recover that.

**Comment by ktoso** [comment]
Great, thanks a lot - I've poked around a bit but it seems the value is being swallowed somewhere... Most likely the osgi plugin since it does a lot of stuff with the manifest.

**Comment by dwijnand** [comment]
Fixed with `OsgiKeys.additionalHeaders += "Multi-Release" -> "true"`, and rebased again.

**Comment by raboof** [comment]
Verified that this indeed produces the `Multi-Release: true` entry in the manifest of akka-actor. Merging into `jdk9-build`, let's iterate more there :)

**Pull request merged**